tools/analyzer_plugin/test/integration/test_bases/server_plugin_contributor_test_base.dart

```diff
@@ -0,0 +1,79 @@
+import 'package:analyzer/src/generated/source.dart';
+import 'package:analyzer/src/generated/source_io.dart';
+import 'package:analyzer_plugin/protocol/protocol_common.dart';
+import 'package:analyzer_plugin/protocol/protocol_generated.dart';
+import 'package:meta/meta.dart';
+import 'package:over_react_analyzer_plugin/src/async_plugin_apis/assist.dart';
+import 'package:over_react_analyzer_plugin/src/diagnostic_contributor.dart';
+import 'package:path/path.dart' as p;
+import 'package:test/test.dart';
+
+import '../mocks.dart';
+import 'analysis_driver_test_base.dart';
+
+abstract class ServerPluginContributorTestBase extends AnalysisDriverTestBase {
+  PluginForTest get testPlugin => _plugin;
+
+  Source applySourceChange(SourceChange change, Source expectedSource) {
+    final path = p.normalize(expectedSource.uri.toFilePath());
+    final applicableFileEdits = change.edits.where((edit) => edit.file == path);
+    expect(applicableFileEdits.map((edit) => edit.file), contains(path),
```

Comment:
> Something about expects in the AssistTestBase class (which isn't really testing anything) rubs me the wrong way. Would you be opposed to replacing these with asserts?

_From @luisgomez-wk on the previous PR_

Reply:
Yeah, I guess I don't have a strong preference here. My thought process here was that in tests, you'd call `applySourceChange()` right before verifying the resulting source contents, so adding another expectation within this method that verifies the given change includes edits for the expected source would be helpful. ¯\_(ツ)_/¯

Reply:
I think checking the edit is applied to the expected source would be helpful!

My worry is that if the expect in this function fails, the test writer will see their test failing, and the stack trace dart gives them will not be in this base class instead of their test

I think an assert would make it clearer that the failure is a side effect of their edits applying on the wrong source

Reply:
TLDR; I think it's fine for now for the sake of merging! And if it does result in confusing failures it'll be easy to change later!